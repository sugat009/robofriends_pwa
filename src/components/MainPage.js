import React, {Component} from 'react';

import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundary from './ErrorBoundry';
import Header from './Header';

import './MainPage.css';

class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    }

    getFilteredRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        });
    };

    render() {
        const {onSearchChange, isPending} = this.props;

        return (
            <div className='tc'>
                <Header/>
                <SearchBox searchChange={onSearchChange}/>
                <Scroll>
                    {isPending ? <h1>Loading</h1> :
                        <ErrorBoundary>
                            <CardList robots={this.getFilteredRobots()}/>
                        </ErrorBoundary>
                    }
                </Scroll>
            </div>
        );
    }

}

export default MainPage;
