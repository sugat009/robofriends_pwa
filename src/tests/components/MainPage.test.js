import React from "react";
import {shallow} from "enzyme";
import MainPage from "../../components/MainPage";

let wrapper, mockProps;

beforeEach(() => {
    mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: "",
        isPending: false
    };
    wrapper = shallow(<MainPage {...mockProps} />);
});

it('should render App Component', function () {
    expect(wrapper).toMatchSnapshot();
});

it('should filter robots correctly', function () {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: "jon",
            email: "jon@gmail.com"
        }],
        searchField: "jon",
        isPending: false
    };
    const wrapper2 = shallow(<MainPage {...mockProps2} />);

    expect(wrapper2.instance().getFilteredRobots()).toEqual([{
        id: 3,
        name: "jon",
        email: "jon@gmail.com"
    }]);
});

it('should filter robots correctly 2', function () {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: "jon",
            email: "jon@gmail.com"
        }],
        searchField: "a",
        isPending: false
    };
    const wrapper3 = shallow(<MainPage {...mockProps3} />);

    expect(wrapper3.instance().getFilteredRobots()).toEqual([]);
});