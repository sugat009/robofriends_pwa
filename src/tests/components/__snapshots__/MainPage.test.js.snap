// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App Component 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
